src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -128,35 +128,192 @@ public Library GetLibrary(LibraryRange libraryRange, NuGetFramework targetFramew
 
             // create a dictionary of dependencies to make sure that no duplicates exist
             var dependencies = new List<LibraryDependency>();
-            TargetFrameworkInformation targetFrameworkInfo = null;
+
+            // Read references from external project
+            if (packageSpec?.RestoreMetadata?.OutputType == RestoreOutputType.NETCore)
+            {
+                // NETCore
+                dependencies.AddRange(GetDependenciesFromSpecRestoreMetadata(packageSpec, targetFramework));
+            }
+            else
+            {
+                // UWP and XProj
```

Comment:
I guess this is until we gut .xproj?

---
In reply to: [82667966](https://github.com/NuGet/NuGet.Client/pull/924#discussion_r82667966) [](ancestors = 82667966)